### PR TITLE
Use Failure to determine problem location

### PR DESCRIPTION
**Pull request by alllex**
A step towards https://github.com/gradle/gradle/issues/28517

Updates `ProblemLocationAnalyzer` implementation to determine the location from stacktrace classified by the `FailureFactory` service, rather than its own flavor of stack trace filitering.

### CODE REVIEW

**Comment by alllex** [comment]
@bot-gradle test qfl

**Comment by alllex** [comment]
@bot-gradle test without pts

**Comment by alllex** [comment]
@bot-gradle test without pts

**Comment by alllex** [comment]
@bot-gradle test apt

File: platforms/ide/problems-api/src/main/java/org/gradle/internal/problems/DefaultProblemDiagnosticsFactory.java [hunk 1]
```diff
@@ -105,7 +109,8 @@ private ProblemDiagnostics locationFromStackTrace(@Nullable Throwable throwable,
         Location location = null;
         if (throwable != null) {
             stackTrace = transformer.transform(throwable.getStackTrace());
-            location = locationAnalyzer.locationForUsage(stackTrace, fromException);
+            Failure stackTracingFailure = failureFactory.create(throwable);
+            location = locationAnalyzer.locationForUsage(stackTracingFailure, fromException);
```
Comment (alllex) [inline platforms/ide/problems-api/src/main/java/org/gradle/internal/problems/DefaultProblemDiagnosticsFactory.java:113]: 💭  This is the main change, temporarily creating a failure to get a classified stacktrace and use it to extract the location.

The follow-up changes would use the created `Failure` and expose it in the `ProblemDiagnostics`

**Review by donat — APPROVED** [review]

**Comment by alllex** [comment]
@bot-gradle merge

**Pull request merged**